pandas/core/nanops.py

```diff
@@ -353,10 +353,16 @@ def _nanvar(values, axis=None, skipna=True, ddof=1):
         values = values.copy()
         np.putmask(values, mask, 0)
 
-    X = _ensure_numeric(values.sum(axis))
-    XX = _ensure_numeric((values ** 2).sum(axis))
-    result = np.fabs((XX - X * X / count) / d)
-    return result
+    # This implementation is based on that of nanvar in numpy 1.8 and up. TODO:
+    # call directly into numpy when we bump numpy up 1.8 .
+    avg = _ensure_numeric(values.sum(axis=axis)) / count
+    if axis is not None:
+        avg = np.expand_dims(avg, axis)
+    sqr = _ensure_numeric((avg - values) ** 2)
+    np.putmask(sqr, mask, 0)
+    var = sqr.sum(axis=axis) / d
```

Comment:
we always specify the ``dtype=`` kw when summing (even for float), because this is the accumulator used. try this with large ``float32`` values and see what happens. (yes this should be on both ``.sum(..)``